File: Programmers/yerim/위장.cpp
// [42578] 위장
#include <string>
#include <vector>
#include <map>
using namespace std;

map<string, int> m;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    for(auto vec: clothes) {
        m[vec[1]]++;
    }
    for(auto info: m) {
        answer *= (info.second + 1);
    }

    return answer - 1;
}

File: Programmers/chaeeun/위장.py
def solution(clothes):
    
    name_list = []
    for i in range(len(clothes)):
        name_list.append(clothes[i][1])

    counts = {}
    for i in name_list:
        try: counts[i] += 1
        except: counts[i] = 1

    answer = 1
    for i in counts.values():
        answer *= (i+1)
    answer -= 1
    # print(answer)
    
    return answer
    
# solution([["crow_mask", "face"], ["blue_sunglasses", "face"], ["smoky_makeup", "face"]])
# solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]])